src/pages/ProposalItem.vue

```diff
@@ -0,0 +1,308 @@
+import { defineComponent, ref, onMounted } from 'vue';
+import { useRoute, useRouter } from 'vue-router';
+import { useQuasar } from 'quasar';
+import JsonViewer from 'vue-json-viewer';
+import { api } from 'src/api';
+import { useAuthenticator } from 'src/composables/useAuthenticator';
+import { RequestedApprovals, Error } from 'src/types';
+
+export default defineComponent({
+  name: 'ProposalItem',
+  components: {
+    JsonViewer: JsonViewer as unknown
+  },
+  setup() {
+    const route = useRoute();
+    const router = useRouter();
+    const $q = useQuasar();
+
+    const { proposalName } = route.params;
+    const { account, getUser } = useAuthenticator();
+
+    const proposer = ref('');
+    const approvalStatus = ref('');
+    const requestedApprovalsRows = ref<RequestedApprovals[]>([]);
+
+    const transactionId = ref('');
+    const transactionData = ref<unknown>({});
+
+    const requestedApprovalsColumns = [
+      {
+        name: 'index',
+        align: 'left',
+        label: '#',
+        field: 'index'
+      },
+      {
+        name: 'actor',
+        align: 'left',
+        label: 'ACTOR',
+        field: 'actor'
+      },
+      {
+        name: 'permission',
+        align: 'left',
+        label: 'PERMISSION',
+        field: 'permission'
+      },
+      {
+        name: 'status',
+        align: 'left',
+        label: 'STATUS',
+        field: 'status'
+      }
+    ];
+
+    onMounted(async () => {
+      try {
+        const {
+          proposals: [proposal]
+        } = await api.getProposals({
+          proposal: proposalName as string,
+          limit: 1
+        });
+
+        if (typeof proposal === 'undefined') {
+          await router.push('/proposal');
+        }
+
+        proposer.value = proposal.proposer;
+        approvalStatus.value = `${proposal.provided_approvals.length}/${
+          proposal.provided_approvals.length +
+          proposal.requested_approvals.length
+        }`;
+
+        let requestedApprovals: RequestedApprovals[] = [];
+
+        proposal.requested_approvals.forEach((item) => {
+          requestedApprovals.push({
+            actor: item.actor,
+            permission: item.permission,
+            status: false,
+            index: 0
+          });
+        });
+
+        proposal.provided_approvals.forEach((item) => {
+          requestedApprovals.push({
+            actor: item.actor,
+            permission: item.permission,
+            status: true,
+            index: 0
+          });
+        });
+
+        requestedApprovalsRows.value = requestedApprovals.map(
+          (item, index) => ({
+            ...item,
+            index: index + 1
+          })
+        );
+
+        const block = await api.getBlock(String(proposal.block_num));
+        transactionId.value = block.transactions[0].trx.id;
+
+        const transaction = await api.getTransaction(transactionId.value);
+        transactionData.value = transaction.actions[0].act.data;
+      } catch (e) {
```

Comment:
refactor the catch error/notify as its own method and pass the exception as an argument (see ** below)

Reply:
Well noticed. Method created!